### PR TITLE
Support -Xexport-library in reverse C interop.

### PR DESCRIPTION
**Pull request by olonho**

### CODE REVIEW

File: backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt [hunk 1]
```diff
@@ -283,7 +283,8 @@ private fun selectExportedLibraries(
 ): List<String> {
     val exportedLibraries = arguments.exportedLibraries?.toList().orEmpty()
 
-    return if (exportedLibraries.isNotEmpty() && outputKind != CompilerOutputKind.FRAMEWORK) {
+    return if (exportedLibraries.isNotEmpty() && outputKind != CompilerOutputKind.FRAMEWORK &&
+            outputKind != CompilerOutputKind.STATIC && outputKind != CompilerOutputKind.DYNAMIC) {
```
Comment (SvyatoslavScherbina) [inline backend.native/cli.bc/src/org/jetbrains/kotlin/cli/bc/K2Native.kt]: Maybe fix error message below too?

File: backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt [hunk 2]
```diff
@@ -639,9 +638,21 @@ internal class CAdapterGenerator(
...
+
+    private val moduleDescriptors = mutableSetOf<ModuleDescriptor>()
+
     fun generateBindings() {
         scopes.push(ExportedElementScope(ScopeKind.TOP, "kotlin"))
-        context.moduleDescriptor.accept(this, null)
+        moduleDescriptors += context.moduleDescriptor
+        moduleDescriptors += context.getExportedDependencies()
+
+        currentPackageFragments = context.moduleDescriptor.getPackageFragments().sortedWith(
```
Comment (SvyatoslavScherbina) [inline backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt]: Maybe include exported dependencies' package fragments too?

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt]: But why, looks like everything is exported anyway (as descriptor tree contains all them).

Comment (SvyatoslavScherbina) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt]: What is `currentPackageFragments` for then?

Comment (olonho) [reply backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/CAdapterGenerator.kt]: OK, reworked.

**Pull request merged**